### PR TITLE
docs(self-hosted): external storage

### PR DESCRIPTION
**Pull request by aldy505**
## DESCRIBE YOUR PR
Dupe of https://github.com/getsentry/develop/pull/1269, as guessed, self-hosted docs is not 100% complete on what kind of thngs that are possible there. Here is one of the PR to help achieve those.

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## LEGAL BOILERPLATE

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

**Comment by aldy505** [comment]
ping @joshuarli @hubertdeng123

**Comment by aldy505** [comment]
another ping @hubertdeng123

File: develop-docs/self-hosted/external-storage.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+You will need to set `GOOGLE_APPLICATION_CREDENTIALS` environment variable. For more information, refer to the [Google Cloud documentation for setting up authentication](https://cloud.google.com/storage/docs/reference/libraries#setting_up_authentication).
+
+```bash
+gs://my-bucket
+```
+
+### S3 backend
+
+<Alert title="Note" level="warning">
+  Although S3 support is available, it is not thoroughly tested and being used by Sentry SaaS internally. Therefore, it is not something that Sentry folks will offer very good support for it.
```
Comment (hubertdeng123) [inline develop-docs/self-hosted/external-storage.mdx:66]:
```suggestion
  Although S3 support is available, it is not thoroughly tested and is not used by Sentry SaaS internally. Therefore, it is experiemental and not officially supported. 
```

File: develop-docs/self-hosted/external-storage.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  endpoint_url: 'https://<REDACTED>' # If you're not using AWS.
+  addressing_style: 'path' # For regular AWS S3, use "auto" or "virtual". For other S3-compatible API like MinIO or Ceph, use "path".
+  signature_version: 's3v4'
+```
+
+Refer to [botocore configuration](https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html) for valid configuration values.
+
+## Vroom
+
+Vroom is the service that handles profiling. By default the data for profiling is saved on local filesystem. On self-hosted deployment, this should be done by overriding the `SENTRY_BUCKET_PROFILES` environment variable. It's also possible that additional environment variables should be added, depending on the backend of choice.
```
Comment (hubertdeng123) [inline develop-docs/self-hosted/external-storage.mdx:53]:
```suggestion
Vroom is the service that handles profiling. By default the data for profiling is saved on the local filesystem. On self-hosted deployments, this should be done by overriding the `SENTRY_BUCKET_PROFILES` environment variable. It's also possible that additional environment variables should be added, depending on the backend of choice.
```

File: develop-docs/self-hosted/external-storage.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+```yaml
+filestore.backend: "gcs"
+filestore.options:
+  bucket_name: "..."
+```
+
+### S3 backend
+
+<Alert title="Warning" level="warning">
+  Although S3 support is available, it is not thoroughly tested and being used by Sentry SaaS internally. Therefore, it is not something that Sentry folks will offer very good support for it.
```
Comment (hubertdeng123) [inline develop-docs/self-hosted/external-storage.mdx:30]:
```suggestion
  Although S3 support is available, it is not thoroughly tested and is not used by Sentry SaaS internally. Therefore, it is experiemental and not officially supported. 
```

File: develop-docs/self-hosted/external-storage.mdx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+### Google Cloud Storage backend
+
+You will need to set `GOOGLE_APPLICATION_CREDENTIALS` environment variable. For more information, refer to the [Google Cloud documentation for setting up authentication](https://cloud.google.com/storage/docs/reference/libraries#setting_up_authentication).
+
+```bash
+gs://my-bucket
+```
+
+### S3 backend
```
Comment (hubertdeng123) [inline develop-docs/self-hosted/external-storage.mdx:63]: I see two S3 backend sections here. Should these be combined?

File: develop-docs/self-hosted/external-storage.mdx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+In some cases, storing Sentry data on-disk is not really something people can do. Sometimes, it's better if they can offload it into some bucket storage (like AWS S3 or Google Cloud Storage).
+
+<Alert title="Note" level="info">
+    After changing configuration files, re-run the <code>./install.sh</code> script, to rebuild and restart the containers. See the <Link to="/self-hosted/#configuration">configuration section</Link> for more information.
+</Alert>
+
+## Sentry
+
+The Sentry service has a abstraction called "filestore" that handles storing attachment, sourcemap (release artifacts), and replays. Filestore configuration for Sentry should be configured on the `sentry/config.yml` file.
```
Comment (hubertdeng123) [inline develop-docs/self-hosted/external-storage.mdx:15]:
```suggestion
Sentry has an abstraction called "filestore" that handles storing attachments, sourcemaps (release artifacts), and replays. Filestore configuration for Sentry should be configured on the `sentry/config.yml` file.
```

File: develop-docs/self-hosted/external-storage.mdx [hunk 6]
```diff
@@ -0,0 +1,86 @@
+---
+title: Self Hosted External Storage
+sidebar_title: External Storage
+sidebar_order: 90
+---
+
+In some cases, storing Sentry data on-disk is not really something people can do. Sometimes, it's better if they can offload it into some bucket storage (like AWS S3 or Google Cloud Storage).
```
Comment (hubertdeng123) [inline develop-docs/self-hosted/external-storage.mdx:7]:
```suggestion
In some cases, storing Sentry data on-disk is not really something people can do. Sometimes, it's better to offload it into some bucket storage (like AWS S3 or Google Cloud Storage).
```

File: develop-docs/self-hosted/external-storage.mdx (hunk 4 above)
Comment (aldy505) [reply develop-docs/self-hosted/external-storage.mdx:63]: No. It's for different services, should resides on a different bucket.

**Comment by aldy505** [comment]
another ping @hubertdeng123 @joshuarli. no more PR on SH docs in the near future from me.

**Comment by aldy505** [comment]
another ping @hubertdeng123 @joshuarli

**Comment by aldy505** [comment]
another feedbacks:

image

image

File: develop-docs/self-hosted/external-storage.mdx [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    GOOGLE_APPLICATION_CREDENTIALS: "/run/secrets/service_account.json"
+  volumes:
+    # The rest of the volumes
+    - "/path/to/service_account.json:/run/secrets/service_account.json:ro"
+```
+
+### S3 backend
+
+<Alert title="Warning" level="warning">
+  Although S3 support is available, it is not thoroughly tested and is not used by Sentry SaaS internally. Therefore, it is experiemental and not officially supported.
```
Comment (aldy505) [inline develop-docs/self-hosted/external-storage.mdx:47]:
```suggestion
  Although S3 support is available, it is not thoroughly tested and is not used by Sentry SaaS internally. Therefore, it is experimental and not officially supported.
```

File: develop-docs/self-hosted/external-storage.mdx [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+      GOOGLE_APPLICATION_CREDENTIALS: "/run/secrets/service_account.json"
+    volumes:
+      # The rest of the volumes
+      - "/path/to/service_account.json:/run/secrets/service_account.json:ro"
+```
+
+### S3 backend
+
+<Alert title="Note" level="warning">
+  Although S3 support is available, it is not thoroughly tested and is not used by Sentry SaaS internally. Therefore, it is experiemental and not officially supported.
```
Comment (aldy505) [inline develop-docs/self-hosted/external-storage.mdx:97]:
```suggestion
  Although S3 support is available, it is not thoroughly tested and is not used by Sentry SaaS internally. Therefore, it is experimental and not officially supported.
```

**Comment by aldy505** [comment]
yet another ping @hubertdeng123 @joshuarli

**Review by hubertdeng123 — APPROVED** [review]

**Pull request merged**